src/main/java/com/sourcegraph/vcs/RepoUtil.java

```diff
@@ -122,39 +123,63 @@ private static String doReplacements(
 
   // Returned format: github.com/sourcegraph/sourcegraph
   // Must be called from non-EDT context
-  private static @NotNull String getRemoteRepoUrlWithoutScheme(
-      @NotNull Project project, @NotNull VirtualFile file) throws Exception {
-    String remoteUrl = getRemoteRepoUrl(project, file);
-    String repoName;
-    try {
-      URL url = new URL(remoteUrl);
-      repoName = url.getHost() + url.getPath();
-    } catch (MalformedURLException e) {
-      repoName = remoteUrl.substring(remoteUrl.indexOf('@') + 1).replaceFirst(":", "/");
-    }
-    return repoName.replaceFirst(".git$", "");
+  private static @NotNull CompletableFuture<String> getRemoteRepoUrlWithoutScheme(
+      @NotNull Project project, @NotNull VirtualFile file) {
+    return getRemoteRepoUrl(project, file)
+        .thenApply(
+            remoteUrl -> {
+              String repoName;
+              try {
+                URL url = new URL(remoteUrl);
+                repoName = url.getHost() + url.getPath();
+              } catch (MalformedURLException e) {
+                repoName = remoteUrl.substring(remoteUrl.indexOf('@') + 1).replaceFirst(":", "/");
+              }
+              return repoName.replaceFirst(".git$", "");
+            });
   }
 
   // Returned format: git@github.com:sourcegraph/sourcegraph.git
   // Must be called from non-EDT context
-  public static @NotNull String getRemoteRepoUrl(
-      @NotNull Project project, @NotNull VirtualFile file) throws Exception {
+  public static @NotNull CompletableFuture<String> getRemoteRepoUrl(
+      @NotNull Project project, @NotNull VirtualFile file) {
     Repository repository = VcsRepositoryManager.getInstance(project).getRepositoryForFile(file);
     VCSType vcsType = getVcsType(project, file);
 
     if (vcsType == VCSType.GIT && repository != null) {
-      return GitUtil.getRemoteRepoUrl((GitRepository) repository, project);
+      String cloneURL;
+      try {
+        cloneURL = GitUtil.getRemoteRepoUrl((GitRepository) repository, project);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+      return CodyAgent.withServer(
+          project,
+          server -> {
+            System.out.println("[debug] cloneURL="+ cloneURL);
+            return server.convertGitCloneURLToCodebaseName(cloneURL).thenApply(result -> {
+              System.out.println("[debug] result="+ result);
```

Comment:
We should use logger here.

(Assuming this is not wip-only/temporary code for debugging and this will end up on main branch - otherwise ignore).

Reply:
I added it only as a temporary "logging" - so, to be removed. I can use a proper logger if we want.